src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -589,6 +590,23 @@ private static void finish(@CheckForNull final String cookie) {
                 return PlaceholderTask.this;
             }
 
+            @Exported
+            public int getNumber() {
+                Run<?, ?> r = getParent().runForDisplay();
+                return r != null ? r.getNumber() : -1;
```

Comment:
Could also declare return type to be `Integer`, in which case `null` will simply skip the entry in the output.